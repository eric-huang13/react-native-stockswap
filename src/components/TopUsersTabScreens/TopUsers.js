import React, {Component} from 'react';
import {
  View,
  TextInput,
  TouchableOpacity,
  ScrollView,
  StyleSheet,
  Text,
  SafeAreaView,
} from 'react-native';
import {connect} from 'react-redux';
import TopUsersPeople from './TopUsersPeople';

import LinearGradient from 'react-native-linear-gradient';
import {moderateScale} from '../../util/responsiveFont'


export class TopUsers extends Component {
  constructor(props) {
    super(props);
    this.state = {
      input: '',
      timeFilter: 'day',
      allPeople: true,
      following: false,
      trending: false,
    };
  }

  timeFilterSelect(time) {
    this.setState({timeFilter: time});
  }
  handleChange = (text) => {
    this.setState({input: text});
  };
  render() {
    const {timeFilter, input} = this.state;
    const {users} = this.props;

    const filteredUsers = users.filter((item) =>
      item.name.toLowerCase().includes(input.toLowerCase()),
    );

    return (
      <LinearGradient
        start={{x: 0.1, y: 0.1}}
        end={{x: 1, y: 1}}
        colors={[
          '#1d2842',
          '#1f2a45',
          '#222d47',
          '#242f4a',
          '#27324d',
          '#293450',
          '#2c3752',
          '#2e3955',
          '#313c58',
          '#333e5c',
          '#36415f',
          '#394463',
        ]}
        style={{flex: 1}}>
        <SafeAreaView>
          {/* <ScrollView> */}
          <LinearGradient
            start={{x: 0.1, y: 0.1}}
            end={{x: 1, y: 1}}
            colors={[
              '#2c3752',
              '#2e3955',
              '#313c58',
              '#333e5c',
              '#36415f',
              '#394463',
            ]}>
            <Text style={style.header}>Top Users</Text>

            <View style={style.tabSelectorContainer}>
              <TouchableOpacity
                onPress={() =>
                  this.setState({
                    allPeople: true,
                    users: false,
                    following: false,
                  })
                }>
                <View
                  style={
                    this.state.allPeople
                      ? {...style.activeTabHeaderView}
                      : {...style.tabHeaderView}
                  }>
                  <Text
                    style={
                      this.state.allPeople
                        ? {...style.activeTabHeader}
                        : {...style.tabHeader}
                    }>
                    All people
                  </Text>
                </View>
              </TouchableOpacity>
              <TouchableOpacity
                onPress={() =>
                  this.setState({
                    allPeople: false,
                    following: true,
                    trending: false,
                  })
                }>
                <View
                  style={
                    this.state.following
                      ? {...style.activeTabHeaderView}
                      : {...style.tabHeaderView}
                  }>
                  <Text
                    style={
                      this.state.following
                        ? {...style.activeTabHeader}
                        : {...style.tabHeader}
                    }>
                    Following
                  </Text>
                </View>
              </TouchableOpacity>
              <TouchableOpacity
                onPress={() =>
                  this.setState({
                    allPeople: false,
                    following: false,
                    trending: true,
                  })
                }>
                <View
                  style={
                    this.state.trending
                      ? {...style.activeTabHeaderView}
                      : {...style.tabHeaderView}
                  }>
                  <Text
                    style={
                      this.state.trending
                        ? {...style.activeTabHeader}
                        : {...style.tabHeader}
                    }>
                    Trending
                  </Text>
                </View>
              </TouchableOpacity>
            </View>
          </LinearGradient>

          {/* //Most likey this will instead be determining which data gets passed into the TopUsersPeople component, unless for some reason we need 3 different components here. I will wait to mess with it until there's actual data coming from backend */}
          {this.state.allPeople ? (
            <TopUsersPeople navigation={this.props.navigation} />
          ) : this.state.following ? (
            <TopUsersPeople navigation={this.props.navigation} />
          ) : this.state.trending ? (
            <TopUsersPeople navigation={this.props.navigation} />
          ) : (
            <View>
              <Text>Search Screen</Text>
            </View>
          )}
          {/* </ScrollView> */}
        </SafeAreaView>
      </LinearGradient>
    );
  }
}

const mapStateToProps = (state) => {
  return {
    users: state.people.users,
  };
};

export default connect(mapStateToProps)(TopUsers);

const style = StyleSheet.create({
  searchInputContainer: {
    alignItems: 'center',
    padding: moderateScale(4.8),
    marginTop: moderateScale(0),
  },
  searchInput: {
    borderWidth: moderateScale(0.5),
    width: '90%',
    fontSize: moderateScale(17),
    fontStyle: 'italic',
    textAlign: 'left',
    padding: moderateScale(2.5),
    opacity: 0.8,
    marginBottom: moderateScale(20),
    paddingLeft: moderateScale(5),
  },
  header: {
    color: '#FFFFFF',
    fontSize: moderateScale(19),
    fontWeight: '700',
    marginVertical: moderateScale(8),
    textAlign: 'center',
  },
  tabSelectorContainer: {
    flexDirection: 'row',
    justifyContent: 'space-around',
    marginVertical: moderateScale(15),
    paddingBottom: moderateScale(2),
  },
  tabHeader: {
    color: '#FFFFFF',
    fontSize: moderateScale(14),
    fontFamily: 'Montserrat-Bold',
  },
  activeTabHeaderView: {
    borderBottomWidth: moderateScale(3),
    borderBottomColor: '#855cff',
    paddingBottom: moderateScale(1.8),
  },
  activeTabHeader: {
    fontFamily: 'Montserrat-Bold',
    color: '#8257FF',
    textShadowColor: '#8257FF',
    textShadowOffset: {width: 1, height: 1},
    textShadowRadius: 50,
    fontSize: moderateScale(14),
  },
});
