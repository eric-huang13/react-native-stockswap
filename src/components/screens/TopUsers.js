  
import React, {Component} from 'react';
import {
  View,
  TextInput,
  TouchableOpacity,
  ScrollView,
  StyleSheet,
  Text,
  SafeAreaView,
} from 'react-native';
import {connect} from 'react-redux';
import TopUsersAllPeople from './TopUsersAllPeople'

import LinearGradient from 'react-native-linear-gradient';


export class TopUsers extends Component {
  constructor(props) {
    super(props);
    this.state = {
      input: '',
      timeFilter: 'day',
      allPeople: true,
      following: false,
      trending: false,
    };
  }

  timeFilterSelect(time) {
    this.setState({timeFilter: time});
  }
  handleChange = (text) => {
    this.setState({input: text});
  };
  render() {
    const {timeFilter, input} = this.state;
    const {users} = this.props;

    const filteredUsers = users.filter((item) =>
      item.name.toLowerCase().includes(input.toLowerCase()),
    );

    return (

<LinearGradient
        start={{x: 0.1, y: 0.1}}
        end={{x: 1, y: 1}}
        colors={[
          '#1d2842',
          '#1f2a45',
          '#222d47',
          '#242f4a',
          '#27324d',
          '#293450',
          '#2c3752',
          '#2e3955',
          '#313c58',
          '#333e5c',
          '#36415f',
          '#394463',
        ]}
        style={{flex: 1}}>
        <SafeAreaView>
          {/* <ScrollView> */}
          <LinearGradient
            start={{x: 0.1, y: 0.1}}
            end={{x: 1, y: 1}}
            colors={[
              '#2c3752',
              '#2e3955',
              '#313c58',
              '#333e5c',
              '#36415f',
              '#394463',
            ]}>
            <Text style={style.header}>Top Users</Text>

            <View style={style.tabSelectorContainer}>
              <TouchableOpacity
                onPress={() =>
                  this.setState({
                    allPeople: true,
                    users: false,
                    following: false,
                  })
                }>
                <View
                  style={
                    this.state.allPeople
                      ? {...style.activeTabHeaderView}
                      : {...style.tabHeaderView}
                  }>
                  <Text
                    style={
                      this.state.allPeople
                        ? {...style.activeTabHeader}
                        : {...style.tabHeader}
                    }>
                    All people
                  </Text>
                </View>
              </TouchableOpacity>
              <TouchableOpacity
                onPress={() =>
                  this.setState({
                    allPeople: false,
                    following: true,
                    trending: false,
                  })
                }>
                <View
                  style={
                    this.state.following
                      ? {...style.activeTabHeaderView}
                      : {...style.tabHeaderView}
                  }>
                  <Text
                    style={
                      this.state.following
                        ? {...style.activeTabHeader}
                        : {...style.tabHeader}
                    }>
                    Following
                  </Text>
                </View>
              </TouchableOpacity>
              <TouchableOpacity
                onPress={() =>
                  this.setState({
                    allPeople: false,
                    following: false,
                    trending: true,
                  })
                }>
                <View
                  style={
                    this.state.trending
                      ? {...style.activeTabHeaderView}
                      : {...style.tabHeaderView}
                  }>
                  <Text
                    style={
                      this.state.trending
                        ? {...style.activeTabHeader}
                        : {...style.tabHeader}
                    }>
                    Trending
                  </Text>
                </View>
              </TouchableOpacity>
            </View>
          </LinearGradient>
          {this.state.allPeople ? (
            <TopUsersAllPeople navigation={this.props.navigation} />
          ) : this.state.following ? (
            <TopUsersAllPeople />
          ) : this.state.trending ? (
            <TopUsersAllPeople navigation={this.props.navigation} />
          ) : (
            <View>
              <Text>Search Screen</Text>
            </View>
          )}
          {/* </ScrollView> */}
        </SafeAreaView>
      </LinearGradient>


      
    );
  }
}

const mapStateToProps = (state) => {
  return {
    users: state.company.users,
  };
};

export default connect(mapStateToProps)(TopUsers);

const style = StyleSheet.create({
  searchInputContainer: {
    alignItems: 'center',
    padding: 4.8,
    marginTop: 0,
  },
  searchInput: {
    borderWidth: 0.5,
    width: '90%',
    fontSize: 17,
    fontStyle: 'italic',
    textAlign: 'left',
    padding: 2.5,
    opacity: 0.8,
    marginBottom: 20,
    paddingLeft: 5,
  },
  header: {
    color: '#FFFFFF',
    fontSize: 19,
    fontWeight: '700',
    marginVertical: 8,
    textAlign: 'center',
  },
  tabSelectorContainer: {
    flexDirection: 'row',
    justifyContent: 'space-around',
    marginVertical: 15,
    paddingBottom: 2,
  },
  tabHeader: {
    color: '#FFFFFF',
    fontSize: 17,
    fontWeight: 'bold',
  },
  activeTabHeaderView: {
    borderBottomWidth: 3,
    borderBottomColor: '#855cff',
    paddingBottom: 1.8,
  },
  activeTabHeader: {
    color: '#8257FF',
    textShadowColor: '#8257FF',
    textShadowOffset: {width: 1, height: 1},
    textShadowRadius: 50,
    fontSize: 17,
    fontWeight: 'bold',
  },

});