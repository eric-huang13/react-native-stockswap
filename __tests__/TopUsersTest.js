import React from 'react';
import TopUsers from '../src/components/TopUsersTabScreens/TopUsers';
import ShallowRenderer from 'react-test-renderer/shallow'; // ES6

jest.mock('@react-native-community/google-signin', () => {});

jest.mock("@react-native-async-storage/async-storage", () =>
  require("@react-native-async-storage/async-storage/jest/async-storage-mock"),
);
jest.mock('react-redux', () => ({
    connect: () => {
      return (component) => {
        return component
      };
    }
  }));
  

test('renders correctly', () => {
    const renderer = new ShallowRenderer();
    renderer.render   (<TopUsers users={[]} />);
    const tree = renderer.getRenderOutput();
  expect(tree).toMatchSnapshot();
});