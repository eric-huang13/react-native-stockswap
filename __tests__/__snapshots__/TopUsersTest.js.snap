// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<LinearGradient
  colors={
    Array [
      "#1d2842",
      "#1f2a45",
      "#222d47",
      "#242f4a",
      "#27324d",
      "#293450",
      "#2c3752",
      "#2e3955",
      "#313c58",
      "#333e5c",
      "#36415f",
      "#394463",
    ]
  }
  end={
    Object {
      "x": 1,
      "y": 1,
    }
  }
  start={
    Object {
      "x": 0.1,
      "y": 0.1,
    }
  }
  style={
    Object {
      "flex": 1,
    }
  }
>
  <ForwardRef(SafeAreaView)>
    <LinearGradient
      colors={
        Array [
          "#2c3752",
          "#2e3955",
          "#313c58",
          "#333e5c",
          "#36415f",
          "#394463",
        ]
      }
      end={
        Object {
          "x": 1,
          "y": 1,
        }
      }
      start={
        Object {
          "x": 0.1,
          "y": 0.1,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#FFFFFF",
            "fontSize": 26.835766423357665,
            "fontWeight": "700",
            "marginVertical": 11.299270072992702,
            "textAlign": "center",
          }
        }
      >
        Top Users
      </Text>
      <View
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "space-around",
            "marginVertical": 21.186131386861312,
            "paddingBottom": 2.8248175182481754,
          }
        }
      >
        <ForwardRef
          onPress={[Function]}
        >
          <View
            style={
              Object {
                "borderBottomColor": "#855cff",
                "borderBottomWidth": 4.237226277372263,
                "paddingBottom": 2.5423357664233577,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#8257FF",
                  "fontFamily": "Montserrat-Bold",
                  "fontSize": 19.77372262773723,
                  "textShadowColor": "#8257FF",
                  "textShadowOffset": Object {
                    "height": 1,
                    "width": 1,
                  },
                  "textShadowRadius": 50,
                }
              }
            >
              All people
            </Text>
          </View>
        </ForwardRef>
        <ForwardRef
          onPress={[Function]}
        >
          <View
            style={Object {}}
          >
            <Text
              style={
                Object {
                  "color": "#FFFFFF",
                  "fontFamily": "Montserrat-Bold",
                  "fontSize": 19.77372262773723,
                }
              }
            >
              Following
            </Text>
          </View>
        </ForwardRef>
        <ForwardRef
          onPress={[Function]}
        >
          <View
            style={Object {}}
          >
            <Text
              style={
                Object {
                  "color": "#FFFFFF",
                  "fontFamily": "Montserrat-Bold",
                  "fontSize": 19.77372262773723,
                }
              }
            >
              Trending
            </Text>
          </View>
        </ForwardRef>
      </View>
    </LinearGradient>
    <TopUsersPeople />
  </ForwardRef(SafeAreaView)>
</LinearGradient>
`;
